assets/js/src/settings/store/actions.ts

```diff
@@ -25,3 +30,138 @@ export function* saveSettings() {
   yield { type: 'TRACK_SETTINGS_SAVED' };
   return { type: 'SAVE_DONE' };
 }
+
+export function* verifyMssKey(key: string, activateMssIfKeyValid: boolean) {
+  const { success, error, res } = yield {
```

Comment:
I have a question here since Redux is new to me. When generators should be used? Is it when any action that may change the state is performed? Is `yield` the way to go through the reducers? Thanks.

And if it is so, why do we also use the API call this way?

Reply:
An [action creator](https://developer.wordpress.org/block-editor/packages/packages-data/#actions) is defined as a generator function when it needs to get some asynchronous data in order to decide which actions to dispatch. The way to get asynchronous data is by `yield`ing a [control](https://developer.wordpress.org/block-editor/packages/packages-data/#controls), so here the `CALL_API` action is a controlled action that calls the API and returns the response, a boolean indicating if the request was successful, and the error if any.